persistence-cassandra/core/src/main/scala/com/lightbend/lagom/internal/persistence/cassandra/ServiceLocatorHolder.scala

```diff
@@ -21,21 +25,36 @@ private[lagom] object ServiceLocatorHolder extends ExtensionId[ServiceLocatorHol
   override def lookup = ServiceLocatorHolder
 
   override def createExtension(system: ExtendedActorSystem): ServiceLocatorHolder =
-    new ServiceLocatorHolder
+    new ServiceLocatorHolder(system)
 }
 
-private[lagom] class ServiceLocatorHolder extends Extension {
+private[lagom] class ServiceLocatorHolder(system: ExtendedActorSystem) extends Extension {
   @volatile private var _serviceLocator: Option[ServiceLocatorAdapter] = None
 
-  def serviceLocator: Option[ServiceLocatorAdapter] = _serviceLocator
+  private val promisedServiceLocator = Promise[ServiceLocatorAdapter]()
+
+  private implicit val exCtx = system.dispatcher
+  private val delayed = akka.pattern.after(10.seconds, using = system.scheduler) {
+    Future.failed(new NoServiceLocatorException("Timed out."))
```

Comment:
I think the error message could be more helpful:

- Log the timeout duration
- Put an explanation in for what might have caused this or how it could be fixed. @jroper said in the original pull request: "It's possible that an end user may misconfigure something, such that a service locator is never provided (particularly in tests). "

So I would suggest something like:

```scala
s"Timed out after $timeout while waiting for a ServiceLocator. Have you configured one?"
```